### ISSUE TITLE
[Export] MeSH invalid terms

### ISSUE THREAD

**Issue by fenekku** [galterlibrary/digital-repository#1042]
If I understood correctly, this is what I extracted to do from @gneidhardt message and it's something Gretchen will do on the DH side

- [x] change on DH the MeSH term "African Continental Ancestry Group/education" to "Blacks/Education" (currently valid MeSH term albeit not a great one there).
- [ ] Fix the MeSH terms that are a variation of "medical education" on DH to to "Education, Medical". Looking up those terms with `"medical education"` breaks DH search, and looking them up with medical education works, but the faceting breaks. In other words, there is 1 or 2 DH bug(s) here, but it can be worked around by not using the facets (not ideal). Exporting MeSH terms in @sharpattack 's librarian export might provide an even better search experience. 

*original message*
I'm not sure if I should add this here, or reopen #1036, but I do have a few issues with the invalid MeSH terms to also report. Let me know if I should post this elsewhere. Luckily fewer MeSH issues than LCSH :)

1. African Continental Ancestry Group/education->FALSE->this has been changed to "Blacks/education" (which is currently being debated - term likely will change again to "Black or African American" in the November 2022 release of new terms, or we might want to introduce a local term "Black people/education"). For now, this affected one record and I changed it to the current term "Blacks/education" though I don't feel great about it.
2. Affecting more records (9 total), I am still having an issue with the various permutations of "Medical education" in the MeSH field. This includes "Education Medical" (https://digitalhub.northwestern.edu/catalog?_=1656517814476&f%5Btags_sim%5D%5B%5D=Education+Medical); "Medical Education" (https://digitalhub.northwestern.edu/catalog?_=1657204766151&f%5Btags_sim%5D%5B%5D=Medical+Education); and "Medical Education " (with trailing space after incorrect text string - https://digitalhub.northwestern.edu/catalog?_=1657204766151&f%5Btags_sim%5D%5B%5D=Medical+Education+). The correct text string should be "Education, Medical". While searching [medical education] (no quotes) does return results, when I use the All Subjects facet and click on the three erroneous text strings (which lead to the three URLs above), I still get an error message. 

The issue with all the medical education errors is that I can't even get to a point to see which records need correcting so that I could look them up another way. If MeSH subjects is a field we can export in the DigitalHub spreadsheet, I could find the records I need to fix that way, and hopefully don't get an error when I look them up via title, DOI, or another method.

_Originally posted by @gneidhardt in https://github.com/galterlibrary/digital-repository/issues/1040#issuecomment-1178105766_

**Comment by gneidhardt** [comment]
Mostly correct, or maybe all correct and I'm slightly misunderstanding your 2 response.

Just to be super clear, 1 is already fixed and validates now since the new term "Blacks" is what was already in the Prism MeSH terms file (though I'd like to put an asterisk there since I think we should revisit, and I think the term will be revised when the Nov 2022 MeSH revision comes out).

2. Even the non-quote search, which initially seems to work, still breaks once I click on the faceted subject for any of those variations. But yes, I think if we can export the subjects as a field in the general DigitalHub export that @sharpattack creates when we enhance metadata then I can isolate the records that have these "medical education" aberrations and hopefully they'll open on DH using the title or DOI. (Sorry if this is what you meant - I just wanted to be super clear that even the non-quote search eventually also returns an error.)

**Comment by fenekku** [comment]
Thanks for the clarifications:

1. is understood to be done and marked as such by the filled checkbox, so that part is good now (I've created: https://github.com/galterlibrary/invenio-subjects-mesh/issues/10 for general term update that should encompass the update to that field)
2. Ah, so the facets were not working on the non-quoted search. So one could go record by record on that search but there is ~1000 records so really not great. Assigning to @sharpattack to see if fixing the facet or exporting MeSH terms in librarian export can be done.

**Comment by Meowcenary** [comment]
@fenekku, @gneidhardt 

Looping back to this now.

From what I'm reading here it seems like we need a way to update the metadata for the records that have some variation on "medical education" as part of their MeSH terms.

I think we could do a few things:

- Use a script to identify the records that need to be updated and then print the ids out
  - Might not be great since there are a lot of records to deal with
- Use a script to actually update the records as we find them. I think this would use basically the same logic as above, but include an update call.
  - We would really need to move carefully since it would mass update metadata. Probably have to do a dry run to just print out what would change and review that before running it fully.
- Update the memoized mesh terms to identify and map the variant terms to the correct mesh id.
  - We would need to figure out what all the variants on this are and probably use a regex or something similar
  
This is how I'm thinking of it now, but if I'm not understanding let me know. Otherwise please let me know your preference on how to approach this.

**Comment by sharpattack** [comment]
There is a faceting issue, but that is separate from main point of this ticket. You have it right @Meowcenary , and your suggestions(using a script or memoized mesh terms) could still be in play. 

@gneidhardt I believe I sent an updated metadata csv with the record ID(let me know if not). This should be enough to directly visit a record page with an invalid "Education, Medical" MeSH term and update it. If it would be better to have one of us automate it, please let us know the invalid term, and what it should be replaced with. 

I am moving the faceting issue into a separate ticket.

**Comment by gneidhardt** [comment]
I was able to directly get to one of the faulty records and correct it, but the "Medical Education" and "Medical Education " were still proving tricky - in the most recent export they are not present in the MeSH column at all. They are however present in the "Keyword" column. Would it be possible that these are mistakenly showing up in the "Subjects" facet and that's why clicking on them leads to an error (since they aren't actually subjects but are keywords)? So in this screenshot you can see them show up in the subjects (which combines LCSH and MeSH and I assume the Geographic and Name subjects).
![Screen Shot 2022-08-18 at 11 17 58 AM](https://user-images.githubusercontent.com/84924248/185444739-9e3fe659-078e-4240-a02b-33c8ccd6ac9e.png)

These are the resulting links that lead to errors, which makes sense because the links are presumably pulling from the subject tags, not the keyword tags: https://digitalhub.northwestern.edu/catalog?_=1660838796940&f%5Btags_sim%5D%5B%5D=Medical+Education and https://digitalhub.northwestern.edu/catalog?_=1660838796940&f%5Btags_sim%5D%5B%5D=Medical+Education+

Though when I paged through the Keywords facet and find the same tags, I also get errors:
 
![Screen Shot 2022-08-18 at 11 21 06 AM](https://user-images.githubusercontent.com/84924248/185445704-18f03b40-b7b4-4ca8-a363-c626e8dabeb7.png)

(URLs are https://digitalhub.northwestern.edu/catalog?_=1660838797336&f%5Btag_sim%5D%5B%5D=Medical+Education and https://digitalhub.northwestern.edu/catalog?_=1660838797336&f%5Btag_sim%5D%5B%5D=Medical+Education+)

I can just to a find search in the CSV and I see can go directly to the collection in DH that have these Keywords - it's a workaround, but I could delete the keyword and add "Education, Medical" to the MeSH headings. Let me know if you'd like me to go ahead and do that or if you'd like me to keep it as is so the problem doesn't disappear. 

For my reference also, here are the affected collections (note that it only seems to matter if the case matches - ie "medical education" in keywords doesn't show up in the subjects list):
https://digitalhub.northwestern.edu/files/655110ec-ae44-4b4e-844d-45838a0717cf
https://digitalhub.northwestern.edu/files/6b476e5f-05a2-46e7-8370-8a4c4aa6e0d5
https://digitalhub.northwestern.edu/files/866e5f43-8d91-4a48-8392-b96f18e34727
https://digitalhub.northwestern.edu/files/ff9e0f47-88be-48f2-932e-5f47d63aa3f7
https://digitalhub.northwestern.edu/files/a775595e-8885-4434-9de6-e52f6e763e8d
https://digitalhub.northwestern.edu/files/80bb5f7b-b8cd-4587-a6fb-7e5817ca9c9f
https://digitalhub.northwestern.edu/files/91eea60f-6a64-4ef1-b8ab-c437c1fc4983

**Comment by sharpattack** [comment]
@gneidhardt The faceting issue has been fixed. The above links should work now(except the "Medical Education " version with an extra space. I removed the trailing extra space from the one record that had it). 

As for the `Medical Education` appearing in both the Keywords and All Subjects facets, it looks like All Subjects includes keywords, MeSH, etc, too. It doesn't mean there is a MeSH value of `Medical Education`. There are still variations of `Education, Medical {something}` MeSH terms though, so I think this issue is still current, right?
